### PR TITLE
Make a python script to check wether all categories are sorted or not

### PR DESCRIPTION
**Pull request by Ammar64**
I made a python script to ensure all categories are sorted.  
Currently all it does is to report unsorted categories.  

it currently outputs this
```
Category Battery is not sorted
Category Browser is not sorted
Category Email Clients is not sorted
Category File Sharing is not sorted
Category Image Manipulation is not sorted
Category Maps & Navigation is not sorted
Category RSS Readers is not sorted
Category System is not sorted
Category ToDo Lists is not sorted
Category URL Manipulation is not sorted
Category Utilities is not sorted
Category Wallpapers is not sorted
Category Weather is not sorted
Category Miscellaneous is not sorted
Category Word is not sorted
```
so check those categories  

TODOS
------------
- [x] make it execute in CI to ensure new PRs are sorted.
- [x] Make it report where it is exactly unsorted.

### CODE REVIEW

**Comment by Ammar64** [comment]
Now outputs this
```
Category Battery is not sorted
    App batterybot pro is not in the correct order
    Should be sorted as follows:
        acca
        battery tool
        batterybot pro
        bcl
        savertuner
Category Browser is not sorted
    App foss browser is not in the correct order
    Should be sorted as follows:
        cromite
        duckduckgo privacy browser
        einkbro
        fennec
        ffupdater
        foss browser
        fulguris web browser
        idiomreplacex browser app
        lightning
        monocles browser
        orbot
        privacy browser
        smartcookieweb
        tor browser
Category Email Clients is not sorted
    App thunderbird is not in the correct order
    Should be sorted as follows:
        fairemail
        k-9 mail
        ltt.rs
        thunderbird
        tutanota
Category File Sharing is not sorted
    App zorin connect is not in the correct order
    Should be sorted as follows:
        cryptomator
        custom uploader
        device connect
        filester
        ftpclient
        kde connect
        localsend
        sharex
        sharik
        sharing
        snapdrop & pairdrop for android
        warpinator
        zorin connect
Category Image Manipulation is not sorted
    App imagepipe is not in the correct order
    Should be sorted as follows:
        easy watermark
        exif thumbnail adder
        ffshare
        image toolbox
        imagepipe
        metadata remover
        privacyblur
        scrambled exif
        superimage
Category Maps & Navigation is not sorted
    App μlogger is not in the correct order
    Should be sorted as follows:
        compass
        gmaps wv
        opentopomap viewer
        organic maps
        osmand~
        private location
        trail sense
        trekarta
        vespucci
        μlogger
Category RSS Readers is not sorted
    App readyou is not in the correct order
    Should be sorted as follows:
        aggregator
        feeder
        handynewsreader
        newsblur
        nunti
        readrops
        readyou
        thud
Category System is not sorted
    App cpu info is not in the correct order
    Should be sorted as follows:
        appmanager
        ashell you
        auditor
        autopie
        busybox installer
        canta
        captive portal controller
        cpu info
        dark theme
        de-bloater
        dhizuku
        droidfs
        dsu sideloader
        geto
        inure app manager (trial)
        libchecker
        logfox
        network survey
        nix-on-droid
        pcapdroid
        permission pilot
        revanced manager
        scoop
        superfreezz
        systemui tuner
        termux
        treble info
        upgradeall
        wifianalyzer
Category ToDo Lists is not sorted
    App todolist is not in the correct order
    Should be sorted as follows:
        1list
        everyday tasks
        opentasks
        orgzly
        pocketplan
        tasks.org
        todolist
Category URL Manipulation is not sorted
    App léon is not in the correct order
    Should be sorted as follows:
        linksheet
        léon
        tarnhelm
        untracker
        untrackme
        urlcheck
Category Utilities is not sorted
    App voice notify is not in the correct order
    Should be sorted as follows:
        amaze file utilities
        avnc
        bendystraw
        bettercounter
        cams
        connectbot
        converter now
        data monitor
        databackup
        deadhash
        etchdroid
        extirpater
        hexviewer
        invizible pro
        json list
        keep alive
        key mapper
        keystoreviewer
        neo backup
        nfcgate
        ocr
        ooni probe
        open video editor
        orgro
        pdf converter
        screen time
        sd maid 2/se
        sensorkraken
        sms import / export
        track & graph
        tts util
        voice notify
        volta
Category Wallpapers is not sorted
    App wallman is not in the correct order
    Should be sorted as follows:
        dark mode live wallpaper
        doodle
        gif live wallpaper
        materials live wallpaper
        palette-cycle
        paperize
        peristyle
        slideshow wallpaper
        wall you
        wallflow
        wallman
        wave lines live wallpaper
        zade's wallpaper
Category Weather is not sorted
    App clima is not in the correct order
    Should be sorted as follows:
        breezy weather
        bura
        cirrus
        clima
        forecastie
        overmorrow weather
        radarweather
        rain
        weather
Category Miscellaneous is not sorted
    App dékadico is not in the correct order
    Should be sorted as follows:
        apple flinger
        ball2box
        burger party
        destination sol
        dooz
        dékadico
        endless sky
        falling blocks
        minetest
        open golf
        pocket broomball
        quinb
        wordlesolver
        xeonjia
        zoysii
Category Word is not sorted
    App lexica is not in the correct order
    Should be sorted as follows:
        acvtool
        android app permissions by izzysoft
        android debloat list
        apatch
        apk editor studio
        apk2url
        apklab
        apktool
        apktoolgui
        awesome android
        cellularprivacy – recommendations
        crosswords
        f-droid
        forkyz
        fossdroid
        guardian project
        gurgle
        izzyondroid f-droid repository
        kernelsu
        lexica
        list of free and open-source android applications
        magisk
        orangefox recovery
        ota on rooted device
        pitch black recovery project
        skyhawk recovery project
        twrp
        universal android debloater gui
```

Is it good ?

**Comment by Ammar64** [comment]
Now it does this
image

**Comment by Ammar64** [comment]
image
it seems like it looks for ensure_sorted.py in this repo instead of the fork.  
Merge and test it.

**Comment by Ammar64** [comment]
Categories name should be written either this way `## – <Category name> –` or this way `### • <Category name>`

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 name: ci
 
-on: [push, pull_request]
+on: [push, pull_request, workflow_dispatch]
```
Comment (offa) [inline .github/workflows/ci.yml:3]: push and pull_request should be sufficient, is there a reason to add workflow_dispatch here?

**Review by offa — APPROVED** [review]

**Comment by offa** [comment]
Nice! :+1: 

The `workflow_dispatch` shouldn't be needed I guess as the others should cover the use cases already.

**Comment by offa** [comment]
So here we go! :smile: 

Thank you! :-)

**Comment by Ammar64** [comment]
@offa unfortunately it still can't find the python file even after merge I wonder why?

**Comment by offa** [comment]
Me too :sweat_smile:

**Comment by offa** [comment]
Ah, checkout is missing …
I'll give it a try.

**Comment by Ammar64** [comment]
Ah string quotations inside `{}` needs python 3.12

**Comment by offa** [comment]
I have another fixed added to `ci` branch.

**Pull request merged**